lib/urn.js

```diff
@@ -12,7 +12,22 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+const MAX_HEX_ID_LENGTH = 65535;
+const RECOGNIZED_PROTOCOLS = new Set([
+  'pbk:ec:secp256r1',
```

Comment:
It would be beneficial to set higher-level category for public keys, like:
`'pbk:ec',
'pbk:rsa'`
Because there are different key sizes and standards, all of them use hex.